### PR TITLE
tools/oomkill: get application lvl stack trace on oomkill

### PR DESCRIPTION
**Pull request by saiaunghlyanhtet**
Get an application level stack trace on oomkill

Related Issue: #3654

### CODE REVIEW

**Review by ekyooo — COMMENTED** [review]
If you change the commit title like below, similar to other tools, it will make it easier to view the git log later. 
"tools/oomkill: ~~"

How about updating oomkill_example.txt to reflect the change in the results?

File: tools/oomkill.py [hunk 1]
```diff
@@ -24,16 +24,19 @@
 bpf_text = """
 #include <uapi/linux/ptrace.h>
 #include <linux/oom.h>
+#include <linux/sched.h>
 
 struct data_t {
     u32 fpid;
     u32 tpid;
     u64 pages;
+    u64 stack_id;
```
[the remark this answers is not in the document]
Comment (saiaunghlyanhtet) [reply tools/oomkill.py]: Made changes as requested.

File: tools/oomkill.py [hunk 2]
```diff
@@ -60,9 +67,15 @@ def print_event(cpu, data, size):
...
+    # Print the stack trace
+    print("  Stack trace:")
+    stack_traces = b["stack_traces"]
+    for addr in stack_traces.walk(event.stack_id):
+        print(f"    {b.sym(addr, event.tpid)}")
+
 # initialize BPF
 b = BPF(text=bpf_text)
-print("Tracing OOM kills... Ctrl-C to stop.")
+print("Tracing OOM kills with stack traces... Ctrl-C to stop.")
```
[the remark this answers is not in the document]
Comment (saiaunghlyanhtet) [reply tools/oomkill.py]: Made changes as requested.

File: tools/oomkill.py [hunk 3]
```diff
@@ -24,16 +24,19 @@
 bpf_text = """
 #include <uapi/linux/ptrace.h>
 #include <linux/oom.h>
+#include <linux/sched.h>
```
[the remark this answers is not in the document]
Comment (saiaunghlyanhtet) [reply tools/oomkill.py]: Made changes as requested.

File: tools/oomkill.py [hunk 4]
```diff
@@ -60,9 +67,15 @@ def print_event(cpu, data, size):
         event.fcomm.decode('utf-8', 'replace'), event.tpid,
         event.tcomm.decode('utf-8', 'replace'), event.pages, avgline))
 
+    # Print the stack trace
+    print("  Stack trace:")
+    stack_traces = b["stack_traces"]
+    for addr in stack_traces.walk(event.stack_id):
```
[the remark this answers is not in the document]
Comment (saiaunghlyanhtet) [reply tools/oomkill.py]: Added handling.

**Pull request merged**